src/radiation/monte_carlo.cpp

```diff
@@ -20,11 +21,95 @@ using Geometry::NDFULL;
 
 namespace radiation {
 
+
 using Microphysics::Opacities;
 
 KOKKOS_INLINE_FUNCTION
 Real GetWeight(const Real wgtC, const Real nu) { return wgtC / nu; }
 
+/**
+ * Integrate Jtot
+ **/
+KOKKOS_INLINE_FUNCTION
+void ComputeTotalEmissivity(Mesh *pmesh) {
+  auto rad = pmesh->packages.Get("radiation");
+  auto opac = pmesh->packages.Get("opacity");
+  const auto nu_min = rad->Param<Real>("nu_min");
+  const auto nu_max = rad->Param<Real>("nu_max");
+  const Real lnu_min = std::log(nu_min);
+  const Real lnu_max = std::log(nu_max);
+  const auto nu_bins = rad->Param<int>("nu_bins");
+  const auto dlnu = rad->Param<Real>("dlnu");
+  const auto nusamp = rad->Param<ParArray1D<Real>>("nusamp");
+  const auto num_particles = rad->Param<int>("num_particles");
+
+  const auto &opacities = opac->Param<Opacities>("opacities");
+  auto species = rad->Param<std::vector<RadiationType>>("species");
+  const auto num_species = rad->Param<int>("num_species");
+  RadiationType species_d[MaxNumRadiationSpecies] = {};
+  for (int s = 0; s < num_species; s++) {
+    species_d[s] = species[s];
+  }
+  // TODO (BLB): pull outer loop into kokkos loop
+  Real Jtot = 0.0;
+  for (auto &pmb : pmesh->block_list) {
+    auto &rc = pmb->meshblock_data.Get();
+    auto geom = Geometry::GetCoordinateSystem(rc.get());
+    const auto coords = pmb->coords;
+
+    auto ib = pmb->cellbounds.GetBoundsI(IndexDomain::interior);
+    auto jb = pmb->cellbounds.GetBoundsJ(IndexDomain::interior);
+    auto kb = pmb->cellbounds.GetBoundsK(IndexDomain::interior);
+
+    namespace p = fluid_prim;
+    PackIndexMap imap;
+    auto v = rc->PackVariables({p::density, p::temperature, p::ye}, imap);
+    const int prho = imap[p::density].first;
+    const int ptemp = imap[p::temperature].first;
+    const int pye = imap[p::ye].first;
+
+    //Real Jtot = 0.0;
+    Real Jtot_b = 0.0; // per block
+    // into par_for
+    pmb->par_reduce(
+        "Radiation::MonteCarlo::Jtot", kb.s, kb.e, 
+        jb.s, jb.e, ib.s, ib.e, 
+        KOKKOS_LAMBDA(const int k, const int j, const int i, Real &result) {
+        const Real gdet = geom.DetGamma(CellLocation::Cent, k, j, i);
+        const Real dx1 = coords.CellWidthFA(X1DIR, k, j, i);
+        const Real dx2 = coords.CellWidthFA(X2DIR, k, j, i);
+        const Real dx3 = coords.CellWidthFA(X3DIR, k, j, i);
+        const Real dens = v(prho, k, j, i);
+        const Real temp = v(ptemp, k, j, i);
+        const Real ye = v(pye, k, j, i);
+        for (int sidx = 0; sidx < num_species; sidx++){
+          auto s = species_d[sidx];
+          for (int n = 0; n <= nu_bins; n++) {
+            Real nu = nusamp(n);
+            Real Jnu = opacities.EmissivityPerNu(dens, temp, ye, s, nu);
+            result += Jnu * gdet * dx1 * dx2 * dx3;
+          } // loop nu
+        } // loop species
+
+        }, Jtot_b); // par_for
+    Jtot += Jtot_b;
+  } // loop block list
+  rad->UpdateParam<Real>("Jtot", reduction::Sum(Jtot));
```

Comment:
It might be worth checking that `Jtot` is what you expect for a homogeneous isotropic gas, like in the thermalization or thin cooling problems.